src/components/RegisterForm/RegisterForm.js

```diff
@@ -26,103 +43,225 @@ class RegisterForm extends Component {
     });
   }
 
-  saveToLocalStore() {
+  setFirstName(val) {
+    this.setState({ firstName: val });
+  }
 
+  setLastName(val) {
+    this.setState({ lastName: val });
   }
 
-  handleSubmit(e) {
-    e.preventDefault();
-    const form = e.target;
-    const data = new FormData(form);
+  setEmail(val) {
+    this.setState({ email: val });
+  }
 
-    const URL = '/api/form-submit-url';
+  setPassword(val) {
+    this.setState({ password: val });
+  }
 
-    fetch(URL, {
-      method: 'POST',
-      body: data
-    });
+  setCompany(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, company: val } });
+  }
+
+  setAddress(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, addressOne: val } });
+  }
+
+  setCity(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, city: val } });
+  }
+
+  setPostalCode(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, postalCode: +val } });
+  }
+
+  setPhone(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, phone: +val } });
+  }
+
+  selectCountry(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, country: val } });
+  }
+
+  selectRegion(val) {
+    const { address } = this.state;
+    this.setState({ address: { ...address, region: val } });
+  }
```

Comment:
Please make a single function that updates state value:
`setAddressValue(value, field)`

Reply:
Well, yes @myrosvas is right, but there is a possibility to do it an easier way to shorten into 1 function as I requested earlier. So we can fix this in the future - and I can show you how.
Good point